test/coverage-tests/src/test/kotlin/org/partiql/test/coverage/RunnerTest.kt

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.partiql.test.coverage
+
+import org.partiql.coverage.api.PartiQLTest
+import org.partiql.coverage.api.PartiQLTestCase
+import org.partiql.coverage.api.PartiQLTestProvider
+import org.partiql.lang.CompilerPipeline
+import org.partiql.lang.eval.Bindings
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.eval.PartiQLResult
+import kotlin.test.Ignore
+import kotlin.test.assertEquals
+
+class RunnerTest {
+
+    @PartiQLTest(provider = SuccessTestProvider::class)
+    fun successTestExample(tc: SuccessTestProvider.ExampleTestCase, result: PartiQLResult.Value) {
+        val value = result.value
+        assertEquals(ExprValueType.BAG, value.type)
+        value.forEach { element ->
+            assertEquals(ExprValueType.STRUCT, element.type)
+        }
+    }
+
+    @PartiQLTest(provider = ComplexTestProvider::class)
+    fun complexTestExample(tc: SimpleTestCase, result: PartiQLResult.Value) {
+        val value = result.value
+        assertEquals(ExprValueType.BAG, value.type)
+        value.forEach { element ->
+            assertEquals(ExprValueType.STRUCT, element.type)
+        }
+    }
+
+    @Ignore
+    @PartiQLTest(provider = InitializationFailureProvider::class)
+    fun initializationFailureExample(tc: InitializationFailureProvider.TestCase, result: ExprValue) {
+        assertEquals(ExprValueType.BOOL, result.type)
+    }
+
+    @Ignore
+    @PartiQLTest(provider = TestInitializationFailureProvider::class)
+    fun testInitializationFailureExample(tc: TestInitializationFailureProvider.TestCase, result: ExprValue) {
+        assertEquals(ExprValueType.STRING, result.type)
+    }
```

Comment:
Just double-checking: do we want to ignore these?

Reply:
Yes, it was to just show that we can use those annotations. Adding a comment to clarify, thanks!